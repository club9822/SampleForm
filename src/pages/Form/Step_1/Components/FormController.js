import React from 'react';
import {StyleSheet, View} from 'react-native';
import SimpleInput from '../../../../components/Input/SimpleInput';
import CalendarInput from '../../../../components/Input/CalendarInput';
import {height} from '../../../../utils/windowDimensions';
import SpinnerButton from '../../../../components/SpinnerButton';
import {connect} from 'react-redux';
import dispatchPayloadToReduxAction from '../../../../actions/dispatchPayloadToReduxAction';
class FormController extends React.PureComponent {
  constructor(props) {
    super(props);
    this.state = {
      name_shop: '',
      name_manager: '',
      id_national: '',
      identification_number: '',
      from: '',
      father_name: '',
      birth_date: '',

      // mobile: '',
      // tel: '',
    };
    this.onButtonPress = this.onButtonPress.bind(this);
  }
  onButtonPress() {
    const validations = require('../../../../utils/validations').default;
    const {isSet, isNumber, maxNumber, minNumber} = validations;

    const {
      name_shop,
      name_manager,
      id_national,
      identification_number,
      from,
      father_name,
      birth_date,
    } = this.state;

    if (
      (isSet(name_shop) &&
        isSet(name_manager) &&
        isSet(id_national) &&
        isSet(identification_number) &&
        isSet(from) &&
        isSet(father_name) &&
        isSet(birth_date)) ||
      true
    ) {
      /**
       *   navigate to second screen
       *   and  save items in redux
       */

      const PAGES = require('../../../../constants/PAGES').PAGES;

      const Navigation = require('react-native-navigation').Navigation;

      try {
        Navigation.push(this.props.componentId, {
          component: {
            id: PAGES.FROM_STEP_2,
            name: PAGES.FROM_STEP_2,
            options: {
              statusBar: {
                backgroundColor: '#4a91ff',
              },
            },
          },
        })
          .then(res => {
            const DISPATCH_FROM_STEP_1_PAYLOAD = require('../../../../types/actionTypes')
              .ACTION_TYPES.DISPATCH_FROM_STEP_1_PAYLOAD;

            this.props.dispatchPayloadToReduxAction({
              type: DISPATCH_FROM_STEP_1_PAYLOAD,
              payload: this.state,
            });

            if (__DEV__) {
              console.log(res);
            }
          })
          .catch(e => {
            if (__DEV__) {
              console.log(e);
            }
          });
      } catch (e) {
        if (__DEV__) {
          console.log(e);
        }
      }
    } else {
    }
  }

  render() {
    return (
      <View style={[styles.container, this.props.containerStyle || {}]}>
        <SimpleInput
          placeholder={'نام صنف'}
          onChangeText={value => {
            this.setState({name_shop: value});
          }}
          value={this.state.name_shop}
          keyboardType={'default'}
        />
        <SimpleInput
          placeholder={'نام پذیرنده'}
          onChangeText={value => {
            this.setState({name_manager: value});
          }}
          value={this.state.name_manager}
          keyboardType={'default'}
        />
        <SimpleInput
          placeholder={'کد ملی'}
          onChangeText={value => {
            this.setState({id_national: value});
          }}
          value={this.state.id_national}
          keyboardType={'decimal-pad'}
        />
        <SimpleInput
          placeholder={'شماره شناسنامه'}
          onChangeText={value => {
            this.setState({identification_number: value});
          }}
          value={this.state.identification_number}
          keyboardType={'decimal-pad'}
        />
        <SimpleInput
          placeholder={'صادره از'}
          onChangeText={value => {
            this.setState({from: value});
          }}
          value={this.state.from}
          keyboardType={'default'}
        />
        <SimpleInput
          placeholder={'نام پدر'}
          onChangeText={value => {
            this.setState({father_name: value});
          }}
          value={this.state.father_name}
          keyboardType={'default'}
        />
        <CalendarInput
          label={'تاریخ تولد'}
          containerStyle={{
            marginTop: height * 0.02,
          }}
          onDaySelect={birth_date => {
            this.setState({birth_date});
          }}
        />

        <SpinnerButton
          label={'مرحله بعد'}
          containerStyle={{
            marginTop: height * 0.15,
          }}
          onPress={this.onButtonPress}
        />
      </View>
    );
  }
}

export default connect(
  null,
  {dispatchPayloadToReduxAction},
)(FormController);
const styles = StyleSheet.create({
  container: {},
});
