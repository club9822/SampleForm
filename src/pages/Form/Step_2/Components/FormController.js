import React from 'react';
import {StyleSheet, View} from 'react-native';
import SimpleInput from '../../../../components/Input/SimpleInput';
import CalendarInput from '../../../../components/Input/CalendarInput';
import {height} from '../../../../utils/windowDimensions';
import SpinnerButton from '../../../../components/SpinnerButton';
import {connect} from 'react-redux';
import dispatchPayloadToReduxAction from '../../../../actions/dispatchPayloadToReduxAction';
import TextAreaInput from '../../../../components/Input/TextAreaInput';
import PickerInput from '../../../../components/Input/PicherInput';
import commonStyles from '../../../../styles/commonStyle';
class FormController extends React.PureComponent {
  constructor(props) {
    super(props);
    this.state = {
      mobile: '',
      tel: '',
      meters: '',
      region_id: 0,
      city_id: 0,
      address: '',
      sub_category_id: 1,
      description: '',
      regions: [
        {id: '1', label: 'تهران', value: 0},
        {id: '1', label: 'تهران', value: 0},
        {id: '1', label: 'تهران', value: 0},
        {id: '1', label: 'تهران', value: 0},
        {id: '1', label: 'تهران', value: 0},
        {id: '1', label: 'تهران', value: 0},
        {id: '1', label: 'تهران', value: 0},
      ],
    };
    this.onButtonPress = this.onButtonPress.bind(this);
  }
  onButtonPress() {
    const validations = require('../../../../utils/validations').default;
    const {isSet, isNumber, maxNumber, minNumber} = validations;

    const {} = this.state;

    if (true) {
      /**
       *   navigate to second screen
       *   and  save items in redux
       */

      const PAGES = require('../../../../constants/PAGES').PAGES;

      const Navigation = require('react-native-navigation').Navigation;

      try {
        Navigation.push(this.props.componentId, {
          component: {
            id: PAGES.FROM_STEP_3,
            name: PAGES.FROM_STEP_3,
            options: {
              statusBar: {
                backgroundColor: '#4a91ff',
              },
            },
          },
        })
          .then(res => {
            const DISPATCH_FROM_PAYLOAD = require('../../../../types/actionTypes')
              .ACTION_TYPES.DISPATCH_FROM_PAYLOAD;

            this.props.dispatchPayloadToReduxAction({
              type: DISPATCH_FROM_PAYLOAD,
              payload: this.state,
            });

            if (__DEV__) {
              console.log(res);
            }
          })
          .catch(e => {
            if (__DEV__) {
              console.log(e);
            }
          });
      } catch (e) {
        if (__DEV__) {
          console.log(e);
        }
      }
    } else {
    }
  }

  render() {
    return (
      <View style={[styles.container, this.props.containerStyle || {}]}>
        <SimpleInput
          placeholder={'شماره همراه'}
          onChangeText={value => {
            this.setState({mobile: value});
          }}
          value={this.state.mobile}
          keyboardType={'decimal-pad'}
        />
        <SimpleInput
          placeholder={'تلفن فروشگاه'}
          onChangeText={value => {
            this.setState({tel: value});
          }}
          value={this.state.tel}
          keyboardType={'decimal-pad'}
        />
        <SimpleInput
          placeholder={'متراژ فروشگاه'}
          onChangeText={value => {
            this.setState({meters: value});
          }}
          value={this.state.meters}
          keyboardType={'decimal-pad'}
        />
        <View
          style={[
            commonStyles.row,
            {width: '85%', marginVertical: height * 0.02, alignSelf: 'center'},
          ]}>
          <PickerInput
            label={'انتخاب محله'}
            data={this.state.regions}
            onItemPress={item => {}}
          />
          <PickerInput
            label={'انتخاب منطقه'}
            data={this.state.regions}
            onItemPress={item => {}}
          />
          <PickerInput
            label={'انتخاب شهر'}
            data={this.state.regions}
            onItemPress={item => {}}
          />
        </View>

        <TextAreaInput
          placeholder={'آدرس دقیق'}
          onChangeText={value => {
            this.setState({description: value});
          }}
          value={this.state.description}
          keyboardType={'default'}
        />

        <View
          style={[
            commonStyles.row,
            {width: '85%', marginVertical: height * 0.02, alignSelf: 'center'},
          ]}>
          <View style={{flex: 3}} />
          <PickerInput
            label={'نوع صنف'}
            data={this.state.regions}
            onItemPress={item => {}}
          />
        </View>
        <TextAreaInput
          placeholder={'توضیحات فروشگاه'}
          onChangeText={value => {
            this.setState({description: value});
          }}
          value={this.state.description}
          keyboardType={'default'}
          containerStyle={{
            height: height * 0.2,
          }}
        />
        <View
          style={[
            commonStyles.row,
            {width: '85%', marginVertical: height * 0.02, alignSelf: 'center'},
          ]}>
          <View style={{flex: 3}} />
          <PickerInput
            label={'درصد تخفیف'}
            data={this.state.regions}
            onItemPress={item => {}}
          />
        </View>
        <View
          style={[
            commonStyles.row,
            {width: '85%', marginVertical: height * 0.01, alignSelf: 'center'},
          ]}>
          <View style={{flex: 3}} />
          <PickerInput
            label={'وضعیت ملکی'}
            data={this.state.regions}
            onItemPress={item => {}}
          />
        </View>
        <View
          style={[
            commonStyles.row,
            {width: '85%', marginVertical: height * 0.01, alignSelf: 'center'},
          ]}>
          <View style={{flex: 3}} />
          <PickerInput
            label={'جواز کسب'}
            data={this.state.regions}
            onItemPress={item => {}}
          />
        </View>
        <SpinnerButton
          label={'مرحله بعد'}
          containerStyle={{
            marginTop: height * 0.1,
          }}
          onPress={this.onButtonPress}
        />
      </View>
    );
  }
}

export default connect(
  null,
  {dispatchPayloadToReduxAction},
)(FormController);
const styles = StyleSheet.create({
  container: {},
});
