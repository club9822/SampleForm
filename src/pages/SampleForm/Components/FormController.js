import React, {Component, useEffect, useState, useReducer} from 'react';
import {
  StyleSheet,
  Platform,
  TouchableWithoutFeedback,
  KeyboardAvoidingView,
  Keyboard,
  View,
} from 'react-native';
import {width, height, isIphoneX} from '../../../utils/windowDimensions';
import {SIGN_IN} from '../../../constants/forms';
//components
import {Fumi} from 'react-native-textinput-effects';
import {
  formReducer,
  initialState,
} from '../../../reducers/formReducer';
import {REDUCER_TYPES} from '../../../types/reducerTypes';
import SpinnerButton from '../../../components/SpinnerButton';
import {ACTION_TYPES} from '../../../types/actionTypes';
import {authActions} from '../../../actions/authActions';
/**
 *
 * @param {Object} props
 * @return {Object}
 * @constructor
 */
// @flow
type Props = {
  containerStyle?: object,
};
function FormController(props: Props) {
  const [state, dispatch] = useReducer(formReducer, initialState);
  const signInAxios = function() {
    const {username, password} = state;

    /**
     * sign action
     */
    authActions({
      type: ACTION_TYPES.SIGN_IN_ACTION,
      payload: {
        username,
        password,
      },
    })
      .then(function(res) {
        alert(res);
        // navigate or any thing
      })
      .catch(function(err) {
        alert(JSON.stringify(err));
      });
  };
  return (
    <TouchableWithoutFeedback onPress={Keyboard.dismiss}>
      <View style={[styles.container, props.containerStyle || {}]}>
        {SIGN_IN.map(function(item, index) {
          return (
            <Fumi
              key={'item__' + index}
              shake={false}
              label={item.label || ''}
              keyboardType={item.keyboardType || 'default'}
              onChangeText={function(value) {
                dispatch({
                  type: REDUCER_TYPES.SIGN_FORM,
                  payload: {
                    key: item.type,
                    value: value,
                  },
                });
              }}
              value={state[`${item.type}`]}
              iconName={item.icon || 'menu'}
            />
          );
        })}
        <SpinnerButton
          label={'ورود'}
          onPress={signInAxios}
          containerStyle={[styles.spinnerButton]}
        />
      </View>
    </TouchableWithoutFeedback>
  );
}

const styles = StyleSheet.create({
  container: {
    justifyContent: 'space-around',
    alignItems: 'center',
    width: '80%',
    height: height * 0.5,
    paddingVertical: height * 0.05,
    borderRadius: width * 0.04,
    ...Platform.select({
      ios: {
        shadowColor: '#000',
        shadowOffset: {
          width: 0,
          height: 2,
        },
        shadowOpacity: 0.25,
        shadowRadius: 3.84,
      },
      android: {
        elevation: height * 0.006,
      },
    }),

    backgroundColor: '#ffffff',
  },
  spinnerButton: {},
});

export default FormController;
