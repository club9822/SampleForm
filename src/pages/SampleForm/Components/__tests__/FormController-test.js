import React from 'react';
import FormController from '../FormController';

import renderer from 'react-test-renderer';

test('renders correctly', () => {
  const tree = renderer.create(<FormController />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders correctly', () => {
  const tree = renderer.create(<FormController containerStyle{{
    marginTop:0
  }} />).toJSON();
  expect(tree).toMatchSnapshot();
});
