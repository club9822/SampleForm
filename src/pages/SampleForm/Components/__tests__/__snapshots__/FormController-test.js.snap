// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onLayout={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#ffffff",
        "borderRadius": 30,
        "height": 667,
        "justifyContent": "space-around",
        "paddingVertical": 66.7,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.25,
        "shadowRadius": 3.84,
        "width": "80%",
      },
      Object {},
    ]
  }
>
  <View
    onLayout={[Function]}
    style={
      Object {
        "alignSelf": "center",
        "backgroundColor": "#fff",
        "height": 127.51470588235294,
        "marginVertical": 4.904411764705882,
        "overflow": "hidden",
        "paddingTop": 0,
        "width": 514.2857142857142,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "color": "#3F5CA6",
            "fontSize": 51.42857142857143,
          },
          Object {
            "alignItems": "center",
            "color": "rgba(163, 163, 163, 1)",
            "height": 58.85294117647059,
            "justifyContent": "center",
            "maxHeight": 58.85294117647059,
            "minHeight": 58.85294117647059,
            "position": "absolute",
            "right": 39.23529411764706,
            "top": 29.426470588235293,
          },
          Object {
            "fontFamily": "Material Design Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#f0f0f0",
            "position": "absolute",
            "width": 1.7,
          },
          Object {
            "height": 127.51470588235294,
            "right": 107.14285714285714,
          },
        ]
      }
    />
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "height": 127.51470588235294,
          "position": "absolute",
          "right": 139.28571428571428,
          "top": 34.330882352941174,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "rgba(148, 148, 148, 1)",
            "fontFamily": "IRANSansMobile(FaNum)",
            "fontSize": 18.857142857142858,
            "height": 58.85294117647059,
            "maxHeight": 58.85294117647059,
            "minHeight": 58.85294117647059,
            "textAlignVertical": "center",
          }
        }
      >
        نام کاربری
      </Text>
    </View>
    <TextInput
      allowFontScaling={true}
      animationDuration={300}
      height={127.51470588235294}
      iconColor="#3F5CA6"
      iconName="account"
      iconSize={51.42857142857143}
      keyboardType="default"
      label="نام کاربری"
      onBlur={[Function]}
      onChange={[Function]}
      onChangeText={[Function]}
      onFocus={[Function]}
      passiveIconColor="#a3a3a3"
      rejectResponderTermination={true}
      selectionColor="#989898"
      shake={false}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "color": "#737373",
            "fontFamily": "IRANSansMobile(FaNum)",
            "height": 98.08823529411764,
            "marginTop": 39.23529411764706,
            "maxHeight": 98.08823529411764,
            "minHeight": 98.08823529411764,
            "textAlign": "right",
            "textAlignVertical": "center",
          },
          Object {
            "color": "#737373",
            "marginLeft": 0,
            "marginRight": 133.92857142857142,
          },
          undefined,
          Object {
            "fontSize": 18.857142857142858,
          },
        ]
      }
      underlineColorAndroid="transparent"
    />
    <View
      style={
        Object {
          "bottom": 19.61764705882353,
          "left": 1,
          "position": "absolute",
        }
      }
    />
  </View>
  <View
    onLayout={[Function]}
    style={
      Object {
        "alignSelf": "center",
        "backgroundColor": "#fff",
        "height": 127.51470588235294,
        "marginVertical": 4.904411764705882,
        "overflow": "hidden",
        "paddingTop": 0,
        "width": 514.2857142857142,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "color": "#3F5CA6",
            "fontSize": 51.42857142857143,
          },
          Object {
            "alignItems": "center",
            "color": "rgba(163, 163, 163, 1)",
            "height": 58.85294117647059,
            "justifyContent": "center",
            "maxHeight": 58.85294117647059,
            "minHeight": 58.85294117647059,
            "position": "absolute",
            "right": 39.23529411764706,
            "top": 29.426470588235293,
          },
          Object {
            "fontFamily": "Material Design Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#f0f0f0",
            "position": "absolute",
            "width": 1.7,
          },
          Object {
            "height": 127.51470588235294,
            "right": 107.14285714285714,
          },
        ]
      }
    />
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "height": 127.51470588235294,
          "position": "absolute",
          "right": 139.28571428571428,
          "top": 34.330882352941174,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "rgba(148, 148, 148, 1)",
            "fontFamily": "IRANSansMobile(FaNum)",
            "fontSize": 18.857142857142858,
            "height": 58.85294117647059,
            "maxHeight": 58.85294117647059,
            "minHeight": 58.85294117647059,
            "textAlignVertical": "center",
          }
        }
      >
        رمز عبور
      </Text>
    </View>
    <TextInput
      allowFontScaling={true}
      animationDuration={300}
      height={127.51470588235294}
      iconColor="#3F5CA6"
      iconName="lock"
      iconSize={51.42857142857143}
      keyboardType="default"
      label="رمز عبور"
      onBlur={[Function]}
      onChange={[Function]}
      onChangeText={[Function]}
      onFocus={[Function]}
      passiveIconColor="#a3a3a3"
      rejectResponderTermination={true}
      selectionColor="#989898"
      shake={false}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "color": "#737373",
            "fontFamily": "IRANSansMobile(FaNum)",
            "height": 98.08823529411764,
            "marginTop": 39.23529411764706,
            "maxHeight": 98.08823529411764,
            "minHeight": 98.08823529411764,
            "textAlign": "right",
            "textAlignVertical": "center",
          },
          Object {
            "color": "#737373",
            "marginLeft": 0,
            "marginRight": 133.92857142857142,
          },
          undefined,
          Object {
            "fontSize": 18.857142857142858,
          },
        ]
      }
      underlineColorAndroid="transparent"
    />
    <View
      style={
        Object {
          "bottom": 19.61764705882353,
          "left": 1,
          "position": "absolute",
        }
      }
    />
  </View>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "center",
        "backgroundColor": "#02cdfc",
        "borderColor": "#665EFF",
        "borderRadius": 32.14285714285714,
        "borderWidth": 0.1,
        "height": 94.16470588235293,
        "justifyContent": "center",
        "marginVertical": 19.61764705882353,
        "opacity": 1,
        "width": "80%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#01111A",
              "fontFamily": "IRANSansMobile(FaNum)",
              "fontSize": 22,
              "textAlign": "center",
            },
            Object {},
          ]
        }
      >
        ورود
      </Text>
    </View>
  </View>
</View>
`;
